src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -18,6 +17,12 @@ public abstract partial class Array : ICloneable, IList, IStructuralComparable,
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern unsafe Array InternalCreate(RuntimeType elementType, int rank, int* pLengths, int* pLowerBounds);
 
+        private static unsafe Array InternalCreateFromArrayType(Type arrayType, int rank, int* pLengths, int* pLowerBounds)
+        {
+            return rank == 1 && (pLowerBounds == null || pLowerBounds[0] == 0)
```

Comment:
This is leaving a lot of perf on the table. `CreateInstanceFromArrayType` should be faster than `CreateInstance` since `CreateInstanceFromArrayType` needs to do less work, but it is the case for this implementation. 

You can micro-benchmark
```
    int[] arg = new int[] { 1, 1 };
    GC.KeepAlive(Array.CreateInstance(typeof(int), arg));
```
vs.
```
    int[] arg = new int[] { 1, 1 };
    GC.KeepAlive(Array.CreateInstanceFromArrayType(typeof(int[,]), arg));
```
`CreateInstanceFromArrayType` is 10% slower on my machine.

Reply:
`CreateInstanceFromArrayType` is about 1.4x faster compared to `CreateInstance` now (CoreCLR, Windows, x64).